debezium-api/src/main/java/io/debezium/spi/converter/RelationalColumn.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.spi.converter;
+
+import java.sql.Types;
+import java.util.Optional;
+
+import io.debezium.common.annotation.Incubating;
+
+/**
+ * A definition of a converted relational column.
+ *
+ * @author Randall Hauch
+
+ */
+@Incubating
+public interface RelationalColumn extends ConvertedField {
+
+    /**
+     * Get the {@link Types JDBC type} for this column
+     *
+     * @return the type constant
+     */
+    int jdbcType();
+
+    /**
+     * Get the database native type for this column
+     *
+     * @return a type constant for the specific database
+     */
+    int nativeType();
+
+    /**
+     * Get the database-specific name of the column's data type.
+     *
+     * @return the name of the type
+     */
+    String typeName();
+
+    /**
+     * Get the database-specific complete expression defining the column's data type, including dimensions, length, precision,
+     * character sets, constraints, etc.
+     *
+     * @return the complete type expression
+     */
+    String typeExpression();
```

Comment:
`Optional<String>`?

Reply:
I don't think so - the type expression is always at least the type name

Reply:
Ok, I was kinda wondering whether it should be empty for a plain name. But works that way.